### DISCUSSION TITLE
Person schema

### DISCUSSION THREAD

**Discussion by Jaikant** [beckn/protocol-specifications#288] [Core Specification]
The person schema, needs a communication attribute. Which can be an array of communication modes.
Communication modes can be phone, whatsapp, telegram, email etc

**Comment by ravi-prakash-v** [comment]
Check out the `Contact` schema in the specification.

```
Contact:
  type: object
  properties:
    phone:
      type: string
    email:
      type: string
    tags:
      $ref: '#/components/schemas/Tags'
```

And `WhatsApp`, `Telegram` etc are technology-specific. The core specification is technology agnostic. May I know what use case is this required for?

If really needed, an additional propoerty called `contact_phone_via` with `type:array` can be added to mention what platforms the phone number is accessible on. But the core spec will keep it as an empty array. Networks may choose to add enums to the `contact_phone_via` property. The possible values for each array item could be `WhatsApp`, `Telegram` etc. Additionally we can also add an array of `phone_communication_modes` with possible values as `TEXT`, `CALL` or `VIDEO-CHAT`.

What do you think?

**Comment by Jaikant** [comment]
We are an e-commerce automation company. Our automations connect to messenger systems as well to automate e-commerce business processes. `contact_phone_via` as an empty array sounds good 👍🏾 
Also `phone_communication_modes` would be excellent to have as outlined above.

**Comment by Jaikant** [comment]
@core-wg-admin As our use cases also takes into account internal organisation structure. The person would need to have a contact(s) as well.

**Comment by venkatramanm** [comment]
Contact via is good option. 👍
